libcxx/include/cmath

```diff
@@ -765,6 +766,54 @@ _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX20 _Tp __constexpr_scalbn(_Tp _
   return __builtin_scalbn(__x, __exp);
 }
 
+#if _LIBCPP_STD_VER >= 17
+
+/// \return the Hermite polynomial \f$ H_n(x) \f$
+/// \note The implementation is based on the recurrence formula
+/// \f[
+///   H_{n+1}(x) = 2x H_n(x) - 2n H_{n-1}
+/// \f]
+/// Press, William H., et al. Numerical recipes 3rd edition: The art of
+/// scientific computing. Cambridge university press, 2007, p. 183.
+template <class _Real>
+_LIBCPP_HIDE_FROM_ABI _Real __hermite(unsigned __n, _Real __x) {
+  if (std::isnan(__x))
+    return std::numeric_limits<_Real>::quiet_NaN();
+
+  _Real __H_0{1};
```

Comment:
Why does the standard make `n >= 128` implementation-defined?

Reply:
I could only speculate. Not sure where to find information on that...

Reply:
Probably somewhere deep within the archives of wg21. I hope @lntue has some insight here.

Reply:
I have no idea why they put that in the standard.  My guess is that some might implement the recurrence relation without multiply-by-2 part, and then scale the final results by `2^n`.  For `n >= 128`, `2^128` exceeds the range of `float` type, so this way of computing hermite polynomials might not support `n >= 128` for `float`.

Reply:
See https://wg21.link/n1884. TLDR: the function changes so rapidly that it's difficult to even get the correct sign.